roles/lib_openshift/src/class/oc_adm_csr.py

```diff
@@ -89,14 +89,15 @@ def match_node(self, csr):
             if node['name'] in self.get_csr_request(csr['spec']['request']):
                 node['csrs'][csr['metadata']['name']] = csr
 
-                # server: check that the username is the node and type is 'Approved'
-                if node['name'] in csr['spec']['username'] and csr['status']:
-                    if csr['status']['conditions'][0]['type'] == 'Approved':
-                        node['server_accepted'] = True
-                # client: check that the username is not the node and type is 'Approved'
-                if node['name'] not in csr['spec']['username'] and csr['status']:
-                    if csr['status']['conditions'][0]['type'] == 'Approved':
-                        node['client_accepted'] = True
+                # client certs may come in as either the service_account or as the node during upgrade
+                # server certs always come in as the node
+                if (node['name'] in csr['spec']['username'] or
```

Comment:
not sure about ansible order of operations but `a || b && c` is typically evaluated `a || (b && c)`, when I think what we want is `(a || b) && c`

Reply:
This is just standard python, you're correct.
```
>>> print a,b,c                                                                                                                                                                                                    
True False False
>>> if (a or b and c): print "true"
... 
true
>>> if (a or b) and c: print "true"
... 
>>> 
```